wheatley/bot.py

```diff
@@ -171,24 +179,43 @@ def look_to_has_been_called(self, call_time: float) -> None:
         self.row_generator = self.next_row_generator or self.row_generator
         self.next_row_generator = None
 
-        # Clear all the flags
+        # Clear all the flags and counters
         self._should_stand = False
-        self._should_start_method = False
         self._should_start_ringing_rounds = False
+        self._rounds_left_before_method = None
+        # Set _rounds_left_before_method if we are ringing up-down-in (3 rounds for backstroke
+        # start; 2 for handstroke)
+        if self._do_up_down_in:
+            self._rounds_left_before_method = 2 if self.row_generator.start_stroke().is_hand() else 3
 
-        # Reset the state, so that the bot starts by ringing rounds
+        # Reset the state, so that Wheatley starts by ringing rounds
         self._is_ringing = True
         self._is_ringing_rounds = True
 
         # Start at the first place of the first row
         self._row_number = 0
         self._place = 0
-        self.start_next_row()
+        self.start_next_row(True)
 
     def _on_go(self) -> None:
         """ Callback called when a user calls 'Go'. """
         if self._is_ringing_rounds:
-            self._should_start_method = True
+            # Calculate how many more rows of rounds we should ring before going into changes (1 if
+            # the person called 'Go' on the same stroke as the RowGenerator starts, otherwise 0).
+            # These values are one less than expected because we are setting
+            # _rounds_left_before_method **after** the row has started.
+            self._rounds_left_before_method = 1 if self.stroke == self.row_generator.start_stroke() else 0
+            # Make sure to call all of the calls that we have missed in the right order (in case the
+            # person calling `Go` called it stupidly late)
+            early_calls = [
+                (ind, calls)
+                for (ind, calls) in self.row_generator.early_calls().items()
+                if ind > self._rounds_left_before_method
+            ]
+            early_calls.sort(key=lambda x: -x[0])
+            for (_index, calls) in early_calls:
+                for c in calls:
```

Comment:
Does there need to be some delay here? This would currently just fire them all, so only the last would be seen properly on screen?

Reply:
> Perhaps a method for this common case

Very good point - I'll do this now.

> Does there need to be some delay here? This would currently just fire them all, so only the last would be seen properly on screen?

Yes, but only for the time being.  Anyway, I can't think of a time where you'd have two calls in a complib composition (`Stand` is removed because it's confusing for this case), but this is not a reason to leave the problem in.

I'd like to replace the `Tower` class with [this library](https://github.com/kneasle/belltower) - it's basically a more complete version of the `Tower` class that I (and anyone else) can use for other projects.  Putting delays between calls is something that makes sense for anyone interacting with RR, so it should live in Belltower.  Therefore I'd make some method like `RingingRoomTower.queue_call`, which would put in the call once enough time had expired since the last one.  But I think `0.6.0` has enough potentially breaking changes, and it'll help debugging if we leave rewriting the internals for after that release.  What do you reckon (also I'd appreciate any PRs/issues for belltower if you feel like making them)?

Reply:
Awesome, I'll check that out properly later. I agree don't try and rush it into 0.6